lib/vdc/CurvilinearGrid.cpp

```diff
@@ -314,28 +315,28 @@ void CurvilinearGrid::ConstCoordItrCG::next()
 
 void CurvilinearGrid::ConstCoordItrCG::next(const long &offset)
 {
-    if (!_index.size()) return;
 
     auto dims = _cg->GetDimensions();
     auto ndims = _cg->GetNumDimensions();
 
-    vector<size_t> maxIndex;
-    maxIndex.reserve(3);
+    if (!ndims) return;
+
+    DimsType maxIndex = {0, 0, 0};
 
-    for (int i = 0; i < ndims; i++) maxIndex.push_back(dims[i] - 1);
+    for (int i = 0; i < ndims; i++) maxIndex[i] = dims[i] - 1;
```

Comment:
There are many instances where `dims[i] - 1` is assigned to another variable of unsigned int type. There's a danger when `dims[i]` holds 0, then nobody knows what `dims[i] - 1` will be. 

I think based on the knowledge of how likely that situation can occur, asserts can be inserted in some instances.

Reply:
Though this is something not related to this PR, so feel free to not deal with it.

Reply:
You can never have to many asserts IMHO :-). Done.